src/main/java/io/cdap/plugin/http/source/common/RawStringPerLine.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2024 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.http.source.common;
+
+
+import io.cdap.plugin.http.common.http.HttpResponse;
+
+import java.io.BufferedReader;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * Class that reads the raw string from the HTTP response and returns it line by line.
+ */
+public class RawStringPerLine implements Closeable, Iterator<String> {
```

Comment:
this class still has some of the issues mentioned before, like it won't honor the number of samples if the first page has fewer rows. 

Let's save that for a follow up though.